src/Jackalope/Node.php

```diff
@@ -1656,8 +1683,9 @@ public function beginTransaction()
         parent::beginTransaction();
 
         // Notify the children properties
-        foreach ($this->properties as $prop) {
-            $prop->beginTransaction();
+        foreach ($this->propertyValues as $name => $propertyValue) {
+            $property = $this->loadProperty($name);
```

Comment:
yes. i guess what you however do need is if you load a property while inside a transaction, pass the transaction status to that property. otherwise the commit/rollback will be a problem on a property that was loaded while inside the transaction.